streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -644,14 +644,15 @@ private int restoreChangelog(final Task task, final ChangelogMetadata changelogM
         final int numRecords = changelogMetadata.bufferedLimitIndex;
 
         if (numRecords != 0) {
-            final List<ConsumerRecord<byte[], byte[]>> records = changelogMetadata.bufferedRecords.subList(0, numRecords);
+            final List<ConsumerRecord<byte[], byte[]>> records = new ArrayList<>(changelogMetadata.bufferedRecords.subList(0, numRecords));
```

Comment:
We'll get concurrent modification for the list exception in L660 `records.size()`. Fixing it by creating a clone of the list, and remove all from the `bufferedRecords` below. We can optimize it later.